src/Proyecto26.RestClient/Helpers/HttpBase.cs

```diff
@@ -46,15 +46,15 @@ public static IEnumerator CreateRequestAndRetry(RequestHelper options, Action<Re
                         callback(null, response);
                         break;
                     }
-                    else if (!options.IsAborted && retries < options.Retries && (IsNetworkError))
+                    else if (!options.IsAborted && retries < options.Retries && (!options.RetryCallbackOnlyOnNetworkErrors || IsNetworkError))
                     {
-                        yield return new WaitForSeconds(options.RetrySecondsDelay);
-                        retries++;
                         if (options.RetryCallback != null)
                         {
                             options.RetryCallback(CreateException(options, request), retries);
                         }
                         DebugLog(options.EnableDebug, string.Format("RestClient - Retry Request\nUrl: {0}\nMethod: {1}", options.Uri, options.Method), false);
+                        yield return new WaitForSeconds(options.RetrySecondsDelay);
+                        retries++;
```

Comment:
Hola bro! Why did you move these lines?

Reply:
Hi! the point of this pull request was to use the retrying callback as a mechanism to repeat a request when failed because of an expired token error. So there is a request that fails because of that, and before retrying again the client notifies of the problem using `RetryCallback` to do the refresh token request and after the delay the client should have had enough time to resolve that request and continue with the original one. That's why I considered moving the callback before the delay

Reply:
Oh ok, looks good to me, thanks!